cmake/Findre2.cmake

```diff
@@ -0,0 +1,103 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+if (re2_FOUND)
+  return()
+endif ()
+
+set(find_package_args)
+if (re2_FIND_VERSION)
+  list(APPEND find_package_args ${re2_FIND_VERSION})
+endif ()
+if (re2_FIND_QUIETLY)
+  list(APPEND find_package_args QUIET)
+endif ()
+set(_vast_cmake_module_path "${CMAKE_MODULE_PATH}")
+unset(CMAKE_MODULE_PATH)
+find_package(re2 ${find_package_args})
+set(CMAKE_MODULE_PATH "${_vast_cmake_module_path}")
+if (re2_FOUND)
+  set(re2_FOUND TRUE)
```

Comment:
I don't get it.

Reply:
Ah, I see this is the module from Arrow. I think you can remove this line.

Reply:
Yeah, definitely. Not sure what they were thinking with this one.